03-pipefilter.md

```diff
@@ -16,277 +16,465 @@ Now that we know a few basic commands,
 
 ~~~ {.bash}
-$ sort -n lengths.txt > sorted-lengths.txt
-$ head -1 sorted-lengths.txt
+$ wc -l *.pdb
 ~~~
 ~~~ {.output}
-  9  methane.pdb
+   30 aldrin.pdb
+    7 ammonia.pdb
+   24 ascorbic-acid.pdb
+   18 benzaldehyde.pdb
+   30 camphene.pdb
+   78 cholesterol.pdb
+   22 cinnamaldehyde.pdb
+   33 citronellal.pdb
+   47 codeine.pdb
+   20 cubane.pdb
+   16 cyclobutane.pdb
+   23 cyclohexanol.pdb
+   13 cyclopropane.pdb
+   12 ethane.pdb
+   13 ethanol.pdb
+   54 ethylcyclohexane.pdb
+   14 glycol.pdb
+   79 heme.pdb
+   15 lactic-acid.pdb
+   49 lactose.pdb
+  248 lanoxin.pdb
+   53 lsd.pdb
+   49 maltose.pdb
+   35 menthol.pdb
+    9 methane.pdb
+   10 methanol.pdb
+   36 mint.pdb
+   44 morphine.pdb
+   41 mustard.pdb
+   33 nerol.pdb
+   51 norethindrone.pdb
+   30 octane.pdb
+   21 pentane.pdb
+   43 piperine.pdb
+   15 propane.pdb
+   25 pyridoxal.pdb
+   52 quinine.pdb
+   51 strychnine.pdb
+   20 styrene.pdb
+   49 sucrose.pdb
+   53 testosterone.pdb
+   42 thiamine.pdb
+   25 tnt.pdb
+   54 tuberin.pdb
+   34 tyrian-purple.pdb
+   23 vanillin.pdb
+   10 vinyl-chloride.pdb
+   55 vitamin-a.pdb
+ 1808 total
 ~~~
 
-Using the parameter `-1` with `head` tells it that
-we only want the first line of the file;
-`-20` would get the first 20,
-and so on.
-Since `sorted-lengths.txt` contains the lengths of our files ordered from least to greatest,
-the output of `head` must be the file with the fewest lines.
-
-If you think this is confusing,
-you're in good company:
-even once you understand what `wc`, `sort`, and `head` do,
-all those intermediate files make it hard to follow what's going on.
-We can make it easier to understand by running `sort` and `head` together:
+Today is our luck day
+and the shell has a command called `sort` that sort the lines of one file.
+Using the pipe we have
 
 ~~~ {.bash}
-$ sort -n lengths.txt | head -1
+$ wc -l *.pdb | sort
 ~~~
 ~~~ {.output}
-  9  methane.pdb
+   10 methanol.pdb
+   10 vinyl-chloride.pdb
+   12 ethane.pdb
+   13 cyclopropane.pdb
+   13 ethanol.pdb
+   14 glycol.pdb
+   15 lactic-acid.pdb
+   15 propane.pdb
+   16 cyclobutane.pdb
+ 1808 total
+   18 benzaldehyde.pdb
+   20 cubane.pdb
+   20 styrene.pdb
+   21 pentane.pdb
+   22 cinnamaldehyde.pdb
+   23 cyclohexanol.pdb
+   23 vanillin.pdb
+  248 lanoxin.pdb
+   24 ascorbic-acid.pdb
+   25 pyridoxal.pdb
+   25 tnt.pdb
+   30 aldrin.pdb
+   30 camphene.pdb
+   30 octane.pdb
+   33 citronellal.pdb
+   33 nerol.pdb
+   34 tyrian-purple.pdb
+   35 menthol.pdb
+   36 mint.pdb
+   41 mustard.pdb
+   42 thiamine.pdb
+   43 piperine.pdb
+   44 morphine.pdb
+   47 codeine.pdb
+   49 lactose.pdb
+   49 maltose.pdb
+   49 sucrose.pdb
+   51 norethindrone.pdb
+   51 strychnine.pdb
+   52 quinine.pdb
+   53 lsd.pdb
+   53 testosterone.pdb
+   54 ethylcyclohexane.pdb
+   54 tuberin.pdb
+   55 vitamin-a.pdb
+   78 cholesterol.pdb
+   79 heme.pdb
+    7 ammonia.pdb
+    9 methane.pdb
 ~~~
 
-The vertical bar between the two commands is called a **pipe**.
```

Comment:
I like this previous explanation - could we move it up to line 143/163 and then after the "we don't have to know or care," maybe refer people to the "Process" callout, if they DO care how it works.